Build failure with http-client 0.5

```
[23 of 25] Compiling Network.PagerDuty.Integration ( src/Network/PagerDuty/Integration.hs, dist/build/Network/PagerDuty/Integration.o )

src/Network/PagerDuty/Integration.hs:295:5: error:
    • Could not deduce (Default Client.Request)
        arising from a use of ‘def’
      from the context: MonadIO m
        bound by the type signature for:
                   submitWith :: MonadIO m =>
                                 Manager -> Logger -> Event -> m (Either Error Response)
        at src/Network/PagerDuty/Integration.hs:(289,1)-(293,39)
    • In the expression: def
      In the second argument of ‘($)’, namely
        ‘def
           {Client.host = "events.pagerduty.com",
            Client.path = "/generic/2010-04-15/create_event.json",
            Client.method = "POST"}’
      In the expression:
        request m l e
        $ def
            {Client.host = "events.pagerduty.com",
             Client.path = "/generic/2010-04-15/create_event.json",
             Client.method = "POST"}
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,7 +24,7 @@ script:
 matrix:
   fast_finish: true
   include:
+    - env: GHCVER=8.0.1
+      addons: { apt: { packages: [ghc-8.0.1], sources: [hvr-ghc] } }
     - env: GHCVER=7.10.3
       addons: { apt: { packages: [ghc-7.10.3], sources: [hvr-ghc] } }
-    - env: GHCVER=7.8.4
-      addons: { apt: { packages: [ghc-7.8.4], sources: [hvr-ghc] } }

--- a/pagerduty.cabal
+++ b/pagerduty.cabal
@@ -72,7 +72,7 @@ library
 
     build-depends:
         aeson                 >= 0.8
-      , base                  >= 4.7      && < 5
+      , base                  >= 4.7 && < 5
       , bifunctors            >= 4.1
       , bytestring            >= 0.9
       , bytestring-conversion >= 0.2.1
@@ -80,7 +80,7 @@ library
       , data-default-class    >= 0.0.1
       , exceptions            >= 0.6
       , generics-sop          >= 0.2.0.0
-      , http-client           >= 0.4.3
+      , http-client           >= 0.5 && < 1
       , http-types            >= 0.8
       , lens                  >= 4.4
       , lens-aeson            >= 1

--- a/script/travis-lifecycle-before_install
+++ b/script/travis-lifecycle-before_install
@@ -2,15 +2,10 @@
 
 set -e
 
-dir=~/.local/bin
-remote=https://github.com/commercialhaskell/stack/releases/download/v0.1.3.1/stack-0.1.3.1-x86_64-linux.gz
-local=$dir/stack
+mkdir -p ~/.local/bin
 
-mkdir -p $dir
-
-curl -L $remote | gunzip > $local
-
-chmod a+x $local
+curl -L https://www.stackage.org/stack/linux-x86_64 | \
+    tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
 
 ghc --version
 stack --version

--- a/src/Control/Monad/Trans/PagerDuty.hs
+++ b/src/Control/Monad/Trans/PagerDuty.hs
@@ -44,7 +44,7 @@ module Control.Monad.Trans.PagerDuty
 import           Control.Applicative
 import           Control.Monad.Base
 import           Control.Monad.Catch
-import           Control.Monad.Error              (MonadError (..))
+import           Control.Monad.Except             (MonadError (..))
 import           Control.Monad.Morph
 import           Control.Monad.Reader
 import           Control.Monad.Trans.Control

--- a/src/Network/PagerDuty/Integration.hs
+++ b/src/Network/PagerDuty/Integration.hs
@@ -81,7 +81,6 @@ module Network.PagerDuty.Integration
 import           Control.Lens
 import           Control.Monad.IO.Class
 import           Data.Aeson
-import           Data.Default.Class
 import qualified Data.HashMap.Strict              as Map
 import           Data.Text                        (Text)
 import           Network.HTTP.Client              (Manager)
@@ -292,7 +291,8 @@ submitWith :: MonadIO m
            -> Event
            -> m (Either Error Response)
 submitWith m l e = request m l e $
-    def { Client.host   = "events.pagerduty.com"
+    Client.defaultRequest
+        { Client.host   = "events.pagerduty.com"
         , Client.path   = "/generic/2010-04-15/create_event.json"
         , Client.method = "POST"
         }

--- a/src/Network/PagerDuty/REST.hs
+++ b/src/Network/PagerDuty/REST.hs
@@ -29,7 +29,6 @@ import           Control.Monad.IO.Class
 import           Control.Monad.Trans
 import           Data.Aeson                       (FromJSON)
 import           Data.Conduit
-import           Data.Default.Class
 import           Data.Monoid
 import           Network.HTTP.Client              (Manager)
 import qualified Network.HTTP.Client              as Client
@@ -90,7 +89,7 @@ http e rq = request (e ^. envManager) (e ^. envLogger) rq $ raw
     }
   where
    raw = case e ^. envAuth of
-        AuthBasic u p -> Client.applyBasicAuth u p def
-        AuthToken t   -> def
+        AuthBasic u p -> Client.applyBasicAuth u p Client.defaultRequest
+        AuthToken t   -> Client.defaultRequest
             { Client.requestHeaders = [("Authorization", "Token token=" <> t)]
             }

--- a/stack-7.10.3.yaml
+++ b/stack-7.10.3.yaml
@@ -1,5 +1,9 @@
-resolver:   lts-5.8
+compiler:   ghc-7.10.3
+resolver:   lts-6.9
 flags:      {}
-extra-deps: []
+extra-deps:
+  - http-conduit-2.2.0
+  - http-client-0.5.0.1
+  - http-client-tls-0.3.0
 packages:
   - '.'

--- a/stack-7.8.4.yaml
+++ b/stack-7.8.4.yaml
@@ -1,6 +1,0 @@
-resolver:   lts-2.21
-flags:      {}
-extra-deps:
-  - generics-sop-0.2.0.0
-packages:
-  - '.'

--- a/stack-8.0.1.yaml
+++ b/stack-8.0.1.yaml
@@ -1,0 +1,6 @@
+compiler:   ghc-8.0.1
+resolver:   nightly-2016-09-18
+flags:      {}
+extra-deps: []
+packages:
+  - '.'

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,1 +1,1 @@
-stack-7.10.3.yaml
+stack-8.0.1.yaml

